haskell-session.el

```diff
@@ -27,6 +27,8 @@
 
 ;;; Code:
 
+(eval-when-compile (require 'cl))
```

Comment:
Please use `cl-lib` instead, as elsewhere in this package. This `require` seems to be for `cl-remove-if` anyway, which is in `cl-lib`. (No `eval-when-compile` is needed/desirable here in that case.)

Reply:
It's actually for the `case` statement; byte compilation kept throwing up semi-spurious errors when I didn't have this line.  I'll try it with `cl-lib` instead.

Reply:
Ah. Then `cl-case` would be the one to use.

Reply:
I've changed it to `cl-lib` now but it seems to fail on `HEAD`; should I change it to just `cl`?

Reply:
That's probably because you've still wrapped it in `eval-when-compile`, which I mentioned above is unnecessary/undesirable.

Reply:
Ahhh, right, just `(require 'cl)` instead of `(require 'cl-lib)`?

Reply:
Nope, `(require 'cl-lib)` instead of  `(eval-when-compile (require 'cl-lib))`.

Reply:
The file already has `(require 'cl-lib)`; but without that `eval-when-compile` statement it throws errors during the CI tests.

Reply:
I don't think that's true. I made a branch with your changes minus that line, and it gives the [same CI results](https://travis-ci.org/haskell/haskell-mode/jobs/626977476) as [master](https://travis-ci.org/haskell/haskell-mode/jobs/614690412). I'll therefore go ahead and merge my branch.

Reply:
Huh, that's weird... either way, thanks for reviewing + merging @purcell !